leakcanary-android-core/src/main/java/leakcanary/DefaultOnHeapAnalyzedListener.kt

```diff
@@ -71,6 +89,44 @@ class DefaultOnHeapAnalyzedListener(private val application: Application) : OnHe
     )
   }
 
+  /**
+   * Android TV devices do not have notifications, therefore the only easy and non-invasive way
+   * to communicate with user is via Toast messages. These are used just to grab user attention and
+   * to direct them to Logcat where a much more detailed report will be printed.
+   */
+  private fun showToast() {
+    // Post the Toast into main thread and wrap it with try-catch in case Toast crashes (it happens)
+    handler.post {
+      try {
+        Toast.makeText(
+            application,
+            "Analysis complete, please check Logcat",
+            Toast.LENGTH_LONG
+        )
+            .show()
+      } catch (exception: Exception) {
+        // Toasts are prone to crashing, ignore
+      }
+    }
+  }
+
+  /**
+   * Android TV with API 26+ has a bug where launcher icon doesn't appear, so users won't know how
+   * to launch LeakCanary Activity.
+   * This method would print an adb command that launched LeakCanary into the logcat
+   */
+  private fun printIntentInfo() {
+    val leakClass = LeakActivity::class.java
+    val message = """====================================
+ANDROID TV LAUNCH INTENT
+====================================
+Use following adb command to open Leak Activity with results:
+
+adb shell am start -n "${application.packageName}/${leakClass.name.replace(leakClass.simpleName, "LeakLauncherActivity")}"
```

Comment:
Why not use LeakActivity directly? The two activities can be launched, no?

Reply:
It can't launch `LeakActivity` - complains about missing permission to do it. It might be because of the activity-alias?